### PR TITLE
add convert_class_to_metric_name func to get rid of hard coding mapping

### PR DESCRIPTION
**Pull request by frankhu-optimizely**
No need to add hardcoded mapping anymore as long as class name follows the pattern like 
`com.optimizely.<module_name>.samza.<task_name>.*`

Signed-off-by: frankhu-optimizely <frank.hu@optimizely.com>

### CODE REVIEW

File: collectors/lib/samza_custom_metric_reporter.py [hunk 1]
```diff
@@ -15,32 +15,51 @@ def __init__(self, consumer_group_id, kafka_bootstrap_servers, kafka_metrics_top
...
+         name_segements[3] != 'samza':
+            return
+
+        # Exception: renmove `Task` at tail
+        metric_name = name_segements[4].lower()
+        if 'task' == metric_name[-4:]:
+            metric_name = metric_name[:-4]
+
+        # Exception: 'com.optimizely.preprocessing.samza.enrichevent.EnrichProjectIdTask' -> 'enrichevents.metrics'
+        if metric_name == 'enrichevent':
```
[the remark this answers is not in the document]
Comment (frankhu-optimizely) [reply collectors/lib/samza_custom_metric_reporter.py]: @mikecdavis will rename cause data loss on dashboards as the tag is changed?

File: collectors/lib/samza_custom_metric_reporter.py [hunk 2]
```diff
@@ -15,32 +15,51 @@ def __init__(self, consumer_group_id, kafka_bootstrap_servers, kafka_metrics_top
...
+        #     'com.optimizely.sessionization.samza.SessionizationTask' : 'sessionization.metrics',
+        #     'com.optimizely.preprocessing.samza.enrichevent.EnrichProjectIdTask' : 'enrichevents.metrics',
+        #     'com.optimizely.preprocessing.samza.enrichevent.processor.EventTicketProjectIdProcessor' : 'enrichevents.metrics',
+        #     'com.optimizely.preprocessing.samza.enrichevent.processor.DecisionEventTicketProjectIdProcessor' : 'enrichevents.metrics',
+        #     'com.optimizely.validator.samza.ValidatorTask': 'validator.metrics',
+        #     'com.optimizely.achievement.samza.AchievementTask': 'achievement.metrics',
+        #
+
+        name_segements = class_name.split('.')
+        if not name_segements or
```
[the remark this answers is not in the document]
Comment (frankhu-optimizely) [reply collectors/lib/samza_custom_metric_reporter.py]: sure, I dunt use regex as we need handle exceptions. if we are going to update dashboard, I prefer to use regex to mk it cleaner

[the remark this answers is not in the document]
Comment (frankhu-optimizely) [reply collectors/lib/samza_custom_metric_reporter.py]: update with regex

File: collectors/lib/samza_custom_metric_reporter.py [hunk 3]
```diff
@@ -15,32 +16,44 @@ def __init__(self, consumer_group_id, kafka_bootstrap_servers, kafka_metrics_top
...
+            ts = int(header_raw['time'] / 1000)
+            tags['source'] = self.sanitize(header_raw['source'])
+
+            for metric_name, metric_val in metric.iteritems():
+                self.print_metrics(
+                    metric_name,
+                    ts,
+                    metric_val,
+                    tags,
+                    convert_class_to_metric_name[class_name])
```
Comment (mikecdavis) [inline collectors/lib/samza_custom_metric_reporter.py]: What happens when this returns `None`? should we be safe and do:
```
metric_name = convert_class_to_metric_name[class_name]
if not metric_name:
  continue

self.print_metrics(...)
```

File: collectors/lib/samza_custom_metric_reporter.py [hunk 4]
```diff
@@ -15,32 +16,44 @@ def __init__(self, consumer_group_id, kafka_bootstrap_servers, kafka_metrics_top
...
+
+            sys.stdout.flush()
+
+    def print_metrics(self, metric_name, ts, value, tags, metric_name_string):
+        if self.is_number(value):
+            print ("%s.%s %d %s %s" % (metric_name_string, metric_name, ts, value, self.to_tsdb_tag_str(tags)))
+
+    def convert_class_to_metric_name(class_name):
+        #   generic func to extract metric_name_string from class_name, compatible with exsiting mapping like
             'com.optimizely.sessionization.samza.SessionizationTask' : 'sessionization.metrics',
```
Comment (mikecdavis) [inline collectors/lib/samza_custom_metric_reporter.py]: we should probably remove all of these... looks like it will fail too.

**Pull request merged**